lib/observability/tracing/tracing.go

```diff
@@ -104,162 +97,44 @@ func (c *Config) CheckAndSetDefaults() error {
 	}
 
-	if c.Client == nil {
-		// first check if a network address is specified, if it was, default
-		// to using grpc. If provided a URL, ensure that it is valid
-		_, _, err := net.SplitHostPort(c.ExporterURL)
-		if err == nil {
-			c.exporterURL = &url.URL{
-				Scheme: "grpc",
-				Host:   c.ExporterURL,
-			}
-		} else {
-			exporterURL, err := url.Parse(c.ExporterURL)
-			if err != nil {
-				return trace.BadParameter("failed to parse exporter URL: %v", err)
-			}
-			c.exporterURL = exporterURL
-		}
-	}
-
-	return nil
-}
-
-func (c *Config) Endpoint() string {
-	if !strings.HasPrefix(c.ExporterURL, c.exporterURL.Scheme) {
-		return c.ExporterURL
-	}
-
-	return c.ExporterURL[len(c.exporterURL.Scheme)+3:]
-}
-
-var _ otlptrace.Client = (*noopClient)(nil)
-
-type noopClient struct{}
-
-func (n noopClient) Start(context.Context) error {
-	return nil
-}
-
-func (n noopClient) Stop(context.Context) error {
-	return nil
-}
-
-func (n noopClient) UploadTraces(context.Context, []*otlp.ResourceSpans) error {
-	return nil
-}
-
-// NewNoopClient returns an oltptrace.Client that does nothing
-func NewNoopClient() otlptrace.Client {
-	return &noopClient{}
-}
-
-// NewStartedClient either returns the provided Config.Client or constructs
-// a new client that is connected to the Config.ExporterURL with the
-// appropriate TLS credentials. The client is started prior to returning to
-// the caller.
-func NewStartedClient(ctx context.Context, cfg Config) (otlptrace.Client, error) {
-	if err := cfg.CheckAndSetDefaults(); err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	clt, err := NewClient(cfg)
-	if err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	ctx, cancel := context.WithTimeout(ctx, cfg.DialTimeout)
-	defer cancel()
-	if err := clt.Start(ctx); err != nil {
-		return nil, trace.Wrap(err)
+	if c.Client != nil {
+		return nil
 	}
 
-	return clt, nil
-}
-
-// NewClient either returns the provided Config.Client or constructs
-// a new client that is connected to the Config.ExporterURL with the
-// appropriate TLS credentials. The returned client is not started,
-// it will be started by the provider if passed to one.
-func NewClient(cfg Config) (otlptrace.Client, error) {
-	if err := cfg.CheckAndSetDefaults(); err != nil {
-		return nil, trace.Wrap(err)
-	}
-
-	if cfg.Client != nil {
-		return cfg.Client, nil
-	}
-
-	var httpOptions []otlptracehttp.Option
-	grpcOptions := []otlptracegrpc.Option{otlptracegrpc.WithDialOption(grpc.WithBlock())}
-
-	if cfg.TLSConfig != nil {
-		httpOptions = append(httpOptions, otlptracehttp.WithTLSClientConfig(cfg.TLSConfig.Clone()))
-		grpcOptions = append(grpcOptions, otlptracegrpc.WithTLSCredentials(credentials.NewTLS(cfg.TLSConfig.Clone())))
-	} else {
-		httpOptions = append(httpOptions, otlptracehttp.WithInsecure())
-		grpcOptions = append(grpcOptions, otlptracegrpc.WithInsecure())
+	// first check if a network address is specified, if it was, default
+	// to using grpc. If provided a URL, ensure that it is valid
```

Comment:
Can't we just demand that a full URL is provided? Is it common for other OTLP things to default to grpc?